kernel/src/table.rs

```diff
@@ -80,6 +81,23 @@ impl Table {
         Snapshot::try_new(self.location.clone(), engine, version)
     }
 
+    /// Create a [`TableChanges`] to get a change data feed for the table between `start_version`,
+    /// and `end_version`. If no `end_version` is supplied, the latest version will be used as the
+    /// `end_version`.
+    pub fn table_changes(
+        &self,
+        engine: &dyn Engine,
+        start_version: Version,
```

Comment:
why not `impl Into<Version>`?

Reply:
Out of curiosity, what situation would produce an `Into<Version>` that is not already `Version`, that we need the fancy arg passing?

Reply:
Yeah, maybe my question should actually have been, "why not `Option<Version>` for `end_version`"? When do we need the into there? I just erred on the side of flexibility

Reply:
Oh using `end_version: impl Into<Option<Version>>` lets us pass in either `Version` or `Option<Version>` to the function. This is a trick I learned from @scovich 
So the following are legal: 
`table.table_changes(engine.as_ref(), 3, 4);`
`table.table_changes(engine.as_ref(), Some(3), 4);`
`table.table_changes(engine.as_ref(), None, 4);`

Reply:
Ahh cool, that's nice. Sorry for the long aside :)